Formatting changes to tests.rst

PoorvajaRayas opened the pull request:
Reviewed documentation and fixed grammatical errors

minrk commented:
This PR now has merge conflicts, due to earlier contributions on the same file. If you resolve the conflicts, we can review this contribution.

If you choose to open a new PR, please close this issue with a link to the new PR.

PoorvajaRayas commented:
@minrk  I have tried to resolve all the conflicts Could you please review it for me?

minrk commented on docs/source/contributing/tests.rst:
```diff
@@ -3,18 +3,17 @@
 
 JupyterHub uses `pytest <https://pytest.org>`_ for all the tests. You
-can find them under the `jupyterhub/tests <https://github.com/jupyterhub/jupyterhub/tree/main/jupyterhub/tests>`_ directory in the git repository.
+can find them under ``jupyterhub/tests``  <https://github.com/jupyterhub/jupyterhub/tree/main/jupyterhub/tests>`_ directory in the git repository.
```
This is rst link syntax. The original was correct.

minrk commented on docs/source/contributing/tests.rst:
```diff
@@ -3,18 +3,17 @@
 ===================================
-
-Unit tests help confirm that JupyterHub works as intended, including after modifications are made. Additionally, they help in clarifying our expectations for our code.
+Unit test helps to confirm that JupyterHub works as it is supposed to work even when the modifications are made . Additionally they clarify us about our expectations from the code. 
```
I don't thihnk the changes here are correct - It should stay `unit tests`. "Works as intended" is a standard phrase we should keep. There also shouldn't be spaces added before periods.

minrk commented on docs/source/contributing/tests.rst:
```diff
@@ -3,18 +3,17 @@
 
 Running the tests
 ==================
+ #Make sure you have completed :ref:`contributing/setup`. Once you are done , you would be able
```
This has lost numbered-list formatting. Please try to preserve the formatting.

No space before punctuation.

minrk commented on docs/source/contributing/tests.rst:
```diff
@@ -3,18 +3,17 @@
 Running the tests
 ==================
+ #Make sure you have completed :ref:`contributing/setup`. Once you are done , you would be able
+ to run ``jupyterhub`` from the command line & access it from your
```
Let's use full words, not `&`

minrk commented on docs/source/contributing/tests.rst:
```diff
@@ -3,18 +3,17 @@
 ==================
+ #Make sure you have completed :ref:`contributing/setup`. Once you are done , you would be able
+ to run ``jupyterhub`` from the command line & access it from your
+ web browser too. This ensures that the dev environment is properly set
```
Remove extra `too` on the end. The last line has been lost (`...up for tests to run`), so this sentence doesn't finish.

minrk commented on docs/source/contributing/tests.rst:
```diff
@@ -54,7 +53,7 @@ Running the tests
       pytest -v jupyterhub/tests/test_api.py::test_shutdown
 
-   For more information, refer to the `pytest usage documentation <https://pytest.readthedocs.io/en/latest/usage.html>`_.
+   For more details refer `pytest usage documentation <https://pytest.readthedocs.io/en/latest/usage.html>`_
```
details is a fine synonym for information, but I'm not sure it's an improvement. The removed comma and `to the` are required for grammar, and should be restored.

minrk commented on docs/source/contributing/tests.rst:
```diff
@@ -95,24 +94,25 @@ And fixtures to add functionality or spawning behavior:
 
-For information on using the existing fixtures and creating new ones, refer to the `pytest fixtures documentation <https://pytest.readthedocs.io/en/latest/fixture.html>`_
-
+Refer `pytest fixtures documentation <https://pytest.readthedocs.io/en/latest/fixture.html>`_ 
```
Missing `to the`

minrk commented on docs/source/contributing/tests.rst:
```diff
@@ -95,24 +94,25 @@ And fixtures to add functionality or spawning behavior:
-For information on using the existing fixtures and creating new ones, refer to the `pytest fixtures documentation <https://pytest.readthedocs.io/en/latest/fixture.html>`_
-
+Refer `pytest fixtures documentation <https://pytest.readthedocs.io/en/latest/fixture.html>`_ 
+to learn how to use fixtures that exists already and also to create new ones
```
Remove unnecessary `also`

minrk commented on docs/source/contributing/tests.rst:
```diff
@@ -95,24 +94,25 @@ And fixtures to add functionality or spawning behavior:
 
-Make sure you have completed all the steps in :ref:`contributing/setup` successfully, and are able to access JupyterHub from your browser at http://localhost:8000 after starting ``jupyterhub`` in your command line.
+Make sure you have completed all the steps in :ref:`contributing/setup` successfully, and
+can launch ``jupyterhub`` from the terminal as well as the web browser
```
It can't be launched from the web browser. The original language had more information - _access_ via the browser _afteR_ launching from the command line. This information should be preserved.

minrk commented on docs/source/contributing/tests.rst:
```diff
@@ -95,24 +94,25 @@ And fixtures to add functionality or spawning behavior:
 JupyterHub has adopted automatic code formatting and linting.
 As long as your code is valid, the pre-commit hook should take care of how it should look.
-You can invoke the pre-commit hook manually at any time with:
+You can invoke the pre-commit hook by hand at any time with :
```
No space before punctuation. I'm not sure about the substitution of manually to its synonym "by hand," can you speak to why you've made that change?

minrk reviewed (commented):
Thanks for your contribution! I’ve left some suggestions inline for you to incorporate. You can update this PR with the suggestions and we will come back to review again.

PoorvajaRayas commented:
> Thanks for your contribution! I’ve left some suggestions inline for you to incorporate. You can update this PR with the suggestions and we will come back to review again.

PoorvajaRayas commented:
> Thanks for your contribution! I’ve left some suggestions inline for you to incorporate. You can update this PR with the suggestions and we will come back to review again.
 Thanks Min for the suggestions . I have reverted most of the changes I had made and just fixed some really small changes .I Please review it for me :)

PoorvajaRayas commented:
I had closed the PR by mistake and Thank you @minrk for the suggestions . I have reverted most of the changes I had made and modified a couple of statements . I would really appreciate another review from you.

minrk commented on docs/source/contributing/tests.rst:
```diff
@@ -11,10 +11,9 @@ can find them under the `jupyterhub/tests <https://github.com/jupyterhub/jupyter
-#. Make sure you have completed :ref:`contributing/setup`. Once completed, you should be able
-   to run ``jupyterhub`` on your command line and access JupyterHub from your browser at http://localhost:8000. Being able to run and access `jupyterhub` should mean that the dev environment is properly set
-   up for tests to run.
+ #Make sure you have completed :ref:`contributing/setup`. Once you are done,you would be able
```
This loses the number-list formatting. Make sure the line still starts with `#. `. Missing space after comma.

minrk commented on docs/source/contributing/tests.rst:
```diff
@@ -11,10 +11,9 @@ can find them under the `jupyterhub/tests <https://github.com/jupyterhub/jupyter
-   to run ``jupyterhub`` on your command line and access JupyterHub from your browser at http://localhost:8000. Being able to run and access `jupyterhub` should mean that the dev environment is properly set
-   up for tests to run.
+ #Make sure you have completed :ref:`contributing/setup`. Once you are done,you would be able
+ to run``jupyterhub``from the command line and access it from your
```
missing space

minrk commented on docs/source/contributing/tests.rst:
```diff
@@ -11,10 +11,9 @@ can find them under the `jupyterhub/tests <https://github.com/jupyterhub/jupyter
-   up for tests to run.
+ #Make sure you have completed :ref:`contributing/setup`. Once you are done,you would be able
+ to run``jupyterhub``from the command line and access it from your
+ web browser.This ensures that the dev environment is properly set up for tests to run.
```
missing space

minrk commented on docs/source/contributing/tests.rst:
```diff
@@ -54,7 +53,7 @@ Running the tests
       pytest -v jupyterhub/tests/test_api.py::test_shutdown
 
-   For more information, refer to the `pytest usage documentation <https://pytest.readthedocs.io/en/latest/usage.html>`_.
+   For more details refer to the `pytest usage documentation <https://pytest.readthedocs.io/en/latest/usage.html>`_
```
The removed punctuation is needed, please put it back.

minrk commented on docs/source/contributing/tests.rst:
```diff
@@ -95,16 +94,15 @@ And fixtures to add functionality or spawning behavior:
 
-For information on using the existing fixtures and creating new ones, refer to the `pytest fixtures documentation <https://pytest.readthedocs.io/en/latest/fixture.html>`_
-
+Refer to the`pytest fixtures documentation <https://pytest.readthedocs.io/en/latest/fixture.html>`_ 
```
Missing space

minrk commented on docs/source/contributing/tests.rst:
```diff
@@ -95,16 +94,15 @@ And fixtures to add functionality or spawning behavior:
-For information on using the existing fixtures and creating new ones, refer to the `pytest fixtures documentation <https://pytest.readthedocs.io/en/latest/fixture.html>`_
-
+Refer to the`pytest fixtures documentation <https://pytest.readthedocs.io/en/latest/fixture.html>`_ 
+to learn how to use fixtures that exists already and to create new ones
```
Missing punctuation

minrk commented on docs/source/contributing/tests.rst:
```diff
@@ -95,16 +94,15 @@ And fixtures to add functionality or spawning behavior:
 -------------------------
-
-Make sure you have completed all the steps in :ref:`contributing/setup` successfully, and are able to access JupyterHub from your browser at http://localhost:8000 after starting ``jupyterhub`` in your command line.
+Make sure you have completed all the steps in :ref:`contributing/setup` successfully, and are able to access JupyterHub from your browser at http://localhost:8000 after starting ``jupyterhub`` in your command line
```
Lost punctuation.

minrk reviewed (changes requested):
Thanks for updating the PR! This now removes lots of punctuation that is actually needed. Can you restore the formatting and punctuation that has been removed?

PoorvajaRayas commented:
@minrk I have made necessary changes as suggested by you.. Please look into it
